File: EvenPositionElement.py
# Given an array of elements print all its elements only at odd positions in the same order seperated by space
n=int(input("Enter range: "))
arr=list(map(int,input().split()))

# method 1

for i in range(n):
    if(i%2==1):
        print(arr[i],end=" ")
'''
iterations: O(n)
T(c)= O(n)
S(c)= O(n)
A(c)= A(1)
'''
#method 2

for i in range(1,n,2):
    print(arr[i],end=" ")

    '''
iterations: O(n/2)
T(c)= O(n)
S(c)= O(n)
A(c)= A(1)
'''

File: EvenPositionElement.java
// Given an array of elements print all its elements only at odd positions in the same order seperated by space

import java.util.Scanner;

public class EvenPositionElement{
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        int arr[]=new int[100];
        System.out.println("Enter n: ");
        int n=sc.nextInt();
        System.out.println("Enter elements: ");
        for(int i=0;i<n;i++){
            int a=sc.nextInt();
            arr[i]=a;
        }
        // method 1

        for(int i =1;i<n;i+=2){
            System.out.print(i+" ");
        }
//iterations: O(n/2)
//T(c)= O(n)
//S(c)= O(n)
        

        //method2

        for(int i =1;i<n;i++){
            if(i%2==1)
            System.out.print(i+" ");
        }
//iterations: O(n)
//T(c)= O(n)
//S(c)= O(n)

    }
}

